tests/test_run.py

```diff
@@ -90,6 +82,80 @@ def test_handle_job_full_execution(synchronous_transitions, db, freezer):
     ]
 
 
+def test_handle_job_full_execution_synchronous(db, freezer):
+    # move to a whole second boundary for easier timestamp maths
+    freezer.move_to("2022-01-01T12:34:56")
+
+    api = StubExecutorAPI()
+    api.synchronous_transitions = [ExecutorState.PREPARING, ExecutorState.FINALIZING]
+
+    start = int(time.time() * 1e9)
```

Comment:
any reason not to use `time_ns`?
```suggestion
    start = int(time.time_ns())
```

Reply:
nope, I'm slowly moveing to time_ns(), I didn't know about it when I originally wrote this.